source/uwp/Renderer/lib/InputValue.cpp

```diff
@@ -537,112 +541,201 @@ try
+}
+
+HRESULT FilteredChoiceSetInputValue::get_CurrentValue(_Outptr_ HSTRING* serializedUserInput)
+try
+{
+    ComPtr<IVector<AdaptiveChoiceInput*>> choices;
+    RETURN_IF_FAILED(m_adaptiveChoiceSetInput->get_Choices(&choices));
+
+    ComPtr<IAdaptiveChoiceInput> selectedChoice;
+    RETURN_IF_FAILED(GetSelectedChoice(&selectedChoice));
+    if (selectedChoice != nullptr)
+    {
+        RETURN_IF_FAILED(selectedChoice->get_Value(serializedUserInput));
+    }
+
+    return S_OK;
+}
+CATCH_RETURN;
 
-    if (choiceSetStyle == ChoiceSetStyle_Compact && !isMultiSelect)
+HRESULT FilteredChoiceSetInputValue::IsValueValid(boolean* isInputValid)
+{
+    boolean isValid;
+
+    // Check if there's text in the autoSuggestBox
+    HString textHString;
+    RETURN_IF_FAILED(m_autoSuggetBox->get_Text(textHString.GetAddressOf()));
+    if (!textHString.IsValid())
     {
-        // Compact style can use the base class implementation
-        RETURN_IF_FAILED(InputValue::SetFocus());
+        // Empty input is only valid if it's not required
+        boolean isRequired;
+        RETURN_IF_FAILED(m_adaptiveInputElement->get_IsRequired(&isRequired));
+        isValid = !isRequired;
     }
     else
     {
-        // For expanded style, put focus on the first choice in the choice set
-        ComPtr<IPanel> panel;
-        RETURN_IF_FAILED(m_uiInputElement.As(&panel));
+        // Non-empty input must match one of the exisiting choices
+        ComPtr<IAdaptiveChoiceInput> selectedChoice;
+        RETURN_IF_FAILED(GetSelectedChoice(&selectedChoice));
+        isValid = (selectedChoice != nullptr);
+    }
+
+    *isInputValid = isValid;
 
-        ComPtr<IVector<UIElement*>> panelChildren;
-        RETURN_IF_FAILED(panel->get_Children(panelChildren.ReleaseAndGetAddressOf()));
+    return S_OK;
+}
+
+HRESULT FilteredChoiceSetInputValue::GetSelectedChoice(IAdaptiveChoiceInput** adaptiveChoiceInput)
+{
+    HString textHString;
+    RETURN_IF_FAILED(m_autoSuggetBox->get_Text(textHString.GetAddressOf()));
+    std::string text = HStringToUTF8(textHString.Get());
 
-        ComPtr<IUIElement> firstChoice;
-        RETURN_IF_FAILED(panelChildren->GetAt(0, &firstChoice));
+    ComPtr<IVector<AdaptiveChoiceInput*>> choices;
+    RETURN_IF_FAILED(m_adaptiveChoiceSetInput->get_Choices(&choices));
 
-        ComPtr<IControl> choiceAsControl;
-        RETURN_IF_FAILED(firstChoice.As(&choiceAsControl));
+    ComPtr<IAdaptiveChoiceInput> selectedChoice;
+    IterateOverVector<AdaptiveChoiceInput, IAdaptiveChoiceInput>(choices.Get(), [&selectedChoice, text](IAdaptiveChoiceInput* choice) {
+        HString titleHString;
+        RETURN_IF_FAILED(choice->get_Title(titleHString.GetAddressOf()));
 
-        boolean isFocused;
-        RETURN_IF_FAILED(choiceAsControl->Focus(FocusState_Programmatic, &isFocused));
-    }
+        std::string title = HStringToUTF8(titleHString.Get());
+
+        if (!ParseUtil::ToLowercase(text).compare(ParseUtil::ToLowercase(title)))
```

Comment:
I don't think we talked about case sensitiveness during the spec review, I believe making the query case insensitive will provide a better user experience, but we may have to raise this detail to achieve parity in all platforms

Reply:
I sent a mail to raise awareness of this.